aptos-move/block-executor/src/captured_reads.rs

```diff
@@ -936,6 +941,62 @@ where
         })
     }
 
+    pub(crate) fn validate_data_reads(
+        &self,
+        data_map: &VersionedData<T::Key, T::Value>,
+        idx_to_validate: TxnIndex,
+    ) -> bool {
+        if self.non_delayed_field_speculative_failure {
+            return false;
+        }
+
+        // This includes AggregatorV1 reads and keeps BlockSTMv1 behavior intact.
+        self.validate_data_reads_impl(self.data_reads.iter(), data_map, idx_to_validate)
+    }
+
+    pub(crate) fn validate_aggregator_v1_reads(
+        &self,
+        data_map: &VersionedData<T::Key, T::Value>,
+        aggregator_write_keys: Option<impl Iterator<Item = T::Key>>,
+        idx_to_validate: TxnIndex,
+    ) -> Result<bool, PanicError> {
+        let mut aggregator_v1_iterable = Vec::with_capacity(self.aggregator_v1_reads.len());
+        for k in &self.aggregator_v1_reads {
+            match self.data_reads.get(k) {
+                Some(data_read) => aggregator_v1_iterable.push((k, data_read)),
+                None => {
+                    return Err(code_invariant_error(format!(
+                        "Aggregator v1 read {:?} not found among captured data reads",
+                        k
+                    )));
+                },
+            }
+        }
+
+        let ret = self.validate_data_reads_impl(
+            aggregator_v1_iterable.into_iter(),
+            data_map,
+            idx_to_validate,
+        );
+
+        if ret && aggregator_write_keys.is_some() {
+            // Additional invariant check (that AggregatorV1 reads are captured for aggregator write keys).
+            // Moreover no extra work when aggregator_write_keys is empty, which is passed for BlockSTMv1.
```

Comment:
What do you mean by `passed for BlockSTMv1`? V1 should not use this function at all?

Reply:
Outdated, another good catch! used to pass these parameters differently I think, now it shouldn't even be called.